pkg/datastore/test/tuples.go

```diff
@@ -598,6 +598,35 @@ func TouchAlreadyExistingTest(t *testing.T, tester DatastoreTester) {
 	ensureTuples(ctx, require, ds, tpl1, tpl2, tpl3)
 }
 
+// WriteDeleteTouchTest tests writing a relationship, deleting it, and then touching it.
+func WriteDeleteTouchTest(t *testing.T, tester DatastoreTester) {
+	require := require.New(t)
+
+	rawDS, err := tester.New(0, veryLargeGCInterval, veryLargeGCWindow, 1)
+	require.NoError(err)
+
+	ds, _ := testfixtures.StandardDatastoreWithData(rawDS, require)
+	ctx := context.Background()
+
+	tpl1 := makeTestTuple("foo", "tom")
+	tpl2 := makeTestTuple("foo", "sarah")
+
+	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_CREATE, tpl1, tpl2)
+	require.NoError(err)
+
+	ensureTuples(ctx, require, ds, tpl1, tpl2)
```

Comment:
Please add a TOUCH here and ensure the tuples are still present

Reply:
@josephschorr 
Does that mean the tests are executed in the order of CREATE, TOUCH, DELETE, TOUCH?

```go
// WriteDeleteTouchTest tests writing a relationship, deleting it, and then touching it.
func WriteDeleteTouchTest(t *testing.T, tester DatastoreTester) {
	require := require.New(t)

	rawDS, err := tester.New(0, veryLargeGCInterval, veryLargeGCWindow, 1)
	require.NoError(err)

	ds, _ := testfixtures.StandardDatastoreWithData(rawDS, require)
	ctx := context.Background()

	tpl1 := makeTestTuple("foo", "tom")
	tpl2 := makeTestTuple("foo", "sarah")

	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_CREATE, tpl1, tpl2)
	require.NoError(err)

	ensureTuples(ctx, require, ds, tpl1, tpl2)

        // Add TOUCH Test
	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_TOUCH, tpl1, tpl2)
	require.NoError(err)

	ensureTuples(ctx, require, ds, tpl1, tpl2)
        // 

	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_DELETE, tpl1, tpl2)
	require.NoError(err)

	ensureNotTuples(ctx, require, ds, tpl1, tpl2)

	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_TOUCH, tpl1, tpl2)
	require.NoError(err)

	ensureTuples(ctx, require, ds, tpl1, tpl2)
}
```

Reply:
Yeah, I'm saying add another test with CREATE, TOUCH, DELETE, TOUCH in addition to one with CREATE, DELETE, TOUCH

Reply:
@josephschorr 
Added test. Could you check it?
[8dae584](https://github.com/authzed/spicedb/pull/1478/commits/8dae584db2e3fa4ad5f4a715c412b3330f060bc4)

Reply:
Running